src/_pytest/recwarn.py

```diff
@@ -336,7 +332,9 @@ def found_str():
 
     @staticmethod
     def _validate_message(wrn: Any) -> None:
-        if not isinstance(msg := wrn.message.args[0], str):
```

Comment:
Can you explain these changes?

Reply:
After making the other changes, the code failed some tests from #11804 that solved #10865, where the incorrect arguments then defaulted to becoming `UserWarning`s. However, the implementation also caught the custom `CustomWarning` class (where `wrn.message.args[0] = int` from `CustomWarning`'s `__init__`, which this looked to fix.
This looks related to #11954

Reply:
Hm, this looks like it could be its own PR potentially. Another implementation: #11959